server.py

```diff
@@ -1,23 +1,116 @@
 import socket
+import email.utils
+import os
+import mimetypes
+import sys
 
-ADDR = ('127.0.0.1', 7000)
-
-socket = socket.socket(
-    socket.AF_INET, socket.SOCK_STREAM, socket.IPPROTO_IP
-    )
-
-socket.bind(ADDR)
-socket.listen(1)
-
-while True:
-    try:
-        conn, addr = socket.accept()
-        while True:
-            msg = conn.recv(16)
-            print msg
-            conn.sendall(msg)
-            if len(msg) < 16:
-                conn.close()
-                break
-    except KeyboardInterrupt:
-        break
\ No newline at end of file
+CRLF = '\r\n'
+ROOT = os.path.relpath('webroot')
+
+
+def server():
+    ADDR = ('127.0.0.1', 10007)
+    socket_ = socket.socket(
+        socket.AF_INET, socket.SOCK_STREAM, socket.IPPROTO_IP
+        )
+    socket_.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
+    socket_.bind(ADDR)
+    socket_.listen(20)
+
+    while True:
+        try:
+            response = ''
+            uri = None
+            uri_info = []
+            try:
+                conn, addr = socket_.accept()
+                uri = parse_request(conn)
+                uri_info = resolve_uri(uri)
+                response = response_ok(uri_info)
+            except (SyntaxError, IndexError):
+                response = response_error(400)
+            except TypeError:
+                response = response_error(406)
+            except NameError:
+                response = response_error(404)
+            except:
+                response = response_error(500)
+            conn.sendall(response)
+            conn.close()
+        except KeyboardInterrupt:
+            break
+
+
+def response_ok(uri_info):
+    returnstr = []
+    date = email.utils.formatdate(usegmt=True)
+    returnstr.append("HTTP/1.1 200 OK")
+    returnstr.append("Date: " + date)
+    returnstr.append("Content-Type: " + uri_info[2])
+    returnstr.append("Content-Length: " + uri_info[1])
+    header = CRLF.join(returnstr)
+    header = header + CRLF + CRLF + uri_info[0]
+    return header
+
+
+def response_error(error):
+    response = "HTTP/1.1 "
+    if error == 400:
+        response += "400 Bad Request"
+    elif error == 404:
+        response = "404 Not Found"
+    elif error == 406:
+        response = "406 Not Acceptable"
+    elif error == 500:
+        response = "500 Server Error"
+    response += CRLF + "Content-Type: text/plain" + CRLF
```

Comment:
You still need two CRLFs after the head section of a response, even if there's no body.